### PR TITLE
allow correct and default ChainID

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: action/actctx.go [hunk 1]
```diff
@@ -48,27 +46,19 @@ func (act *AbstractAction) GasPrice() *big.Int {
...
 		size += len(act.gasPrice.Bytes())
 	}
 
 	return uint32(size)
 }
 
-// SetEnvelopeContext sets the SealedEnvelope context to action context.
-func (act *AbstractAction) SetEnvelopeContext(selp SealedEnvelope) {
+// SetEnvelopeContext sets the Envelope context to action context.
+func (act *AbstractAction) SetEnvelopeContext(ab AbstractAction) {
```
Comment (dustinxie) [inline action/actctx.go:57]: now we don't need the entire SealedEnvelope, only need the AbstractAction part

File: action/envelope.go [hunk 2]
```diff
@@ -84,6 +83,7 @@ func (elp *envelope) Proto() *iotextypes.ActionCore {
 		Version:  elp.version,
 		Nonce:    elp.nonce,
 		GasLimit: elp.gasLimit,
+		ChainID:  elp.chainID,
```
Comment (dustinxie) [inline action/envelope.go:86]: this is missing in earlier implementation

File: action/envelope.go [hunk 3]
```diff
@@ -138,6 +138,7 @@ func (elp *envelope) LoadProto(pbAct *iotextypes.ActionCore) error {
 	*elp = envelope{}
 	elp.version = pbAct.GetVersion()
 	elp.nonce = pbAct.GetNonce()
+	elp.chainID = pbAct.GetChainID()
```
Comment (dustinxie) [inline action/envelope.go:141]: missing in earlier implementation

File: api/coreservice.go [hunk 4]
```diff
@@ -265,10 +265,10 @@ func (core *coreService) SendAction(ctx context.Context, in *iotextypes.Action)
 	}
 
 	// reject action if chainID is not matched at KamchatkaHeight
-	if core.cfg.Genesis.Blockchain.IsToBeEnabled(core.bc.TipHeight()) {
-		if core.bc.ChainID() != in.GetCore().GetChainID() {
-			return "", status.Errorf(codes.InvalidArgument, "ChainID does not match, expecting %d, got %d", core.bc.ChainID(), in.GetCore().GetChainID())
-		}
+	chainID := in.GetCore().GetChainID()
+	if core.cfg.Genesis.Blockchain.IsToBeEnabled(core.bc.TipHeight()) &&
+		chainID != core.bc.ChainID() && chainID != 0 {
```
Comment (dustinxie) [inline api/coreservice.go:270]: as discussed, step 1 we allow correct chainID and default = 0

**Review by CoderZhi — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

**Pull request merged**